### ISSUE TITLE
Automate the renewal of the cdn.httparchive.org ssl cert

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/httparchive.org#14]
- [X] Run: `sudo certbot certonly --manual --preferred-challenges dns`
- [X] Update DNS TXT record with challenge text on Google Domains
- [X] Create a new certificate on GCP and assign to the load balancer HTTPS frontend

**Comment by rviscomi** [comment]
GCS [no longer allows ACME challenges](https://cloud.google.com/storage/docs/naming#objectnames). Need to use `--preferred-challenges dns` to verify domain ownership with DNS instead.

**Comment by rviscomi** [comment]
Cert expiring again on June 22. Kicking the can down the road until it can be automated.

**Comment by rviscomi** [comment]
Updated certificate and setting reminder to renew it again before September 19, 2018.

**Comment by rviscomi** [comment]
Updated cert. We'll need to do this again on December 16, 2018.

**Comment by rviscomi** [comment]
Renewed. Good until March 5, 2019.

Helpful command for checking if the new DNS setting has taken effect: 

`dig -t txt _acme-challenge.cdn.httparchive.org`

**Comment by rviscomi** [comment]
Renewed until April 3

**Comment by rviscomi** [comment]
Reopening this issue and retasking it to automate the renewal of the cert. We can't keep punting it every 3 months.

**Comment by rviscomi** [comment]
Cert expires on June 3

**Comment by tjmonsi** [comment]
Is there a process on renewing this?

**Comment by rviscomi** [comment]
@tjmonsi yes, the manual process is documented in https://github.com/HTTPArchive/httparchive.org/issues/14#issue-286990392

**Comment by tjmonsi** [comment]
I see, I was thinking we were using Cloud DNS. I do remember creating an automated script for a wordpress server running on GCP. 

So if i understood it right, our servers are running on App Engine, and we are connecting a DNS record to it.

**Comment by rviscomi** [comment]
You're right that we have a DNS record pointing the httparchive.org domain name to App Engine.

There are two other DNS settings relevant to this issue: a record pointing cdn.httparchive.org to the GCP front end loader balancer for Cloud Storage, and a TXT record used to verify ownership during certificate renewals.

Dynamically updating the TXT record is the only thing preventing us from automating the renewal process.

**Comment by tomayac** [comment]
@rviscomi This is hitting us now, the certificate for https://cdn.httparchive.org expired on June 3.
![Screen Shot 2019-06-04 at 09 01 59](https://user-images.githubusercontent.com/145676/58858262-790bfd00-86a7-11e9-8a67-ce399e046698.png)

**Comment by rviscomi** [comment]
Cert updated. Sorry about the lapse 😞

**Comment by rviscomi** [comment]
The cert is scheduled to expire on September 2, so I tried to get the Google-managed certificate working again, and it seems like it's holding.

![image](https://user-images.githubusercontent.com/1120896/63455671-38f12680-c41b-11e9-970d-25293536b840.png)

According to https://cloud.google.com/load-balancing/docs/ssl-certificates#google-managed_ssl_certificate_renewal the cert should automatically renew before its expiration on November 19. I've set a reminder to double check this on the 18th.

Hopefully this resolves the issue. Leaving it open until November 18. 🤞
